test/define-test.js

```diff
@@ -244,14 +243,12 @@ dev.devOnlyTest('Only can-type error should be catched', function(assert) {
 	}
 
 	var t = new T();
+	t.aStr = null;
 
-	var obs = new Observation(function() {
```

Comment:
Without this observation the other error is not thrown, right? Since `this.aStr.toUpperCase();` will not be called again. I think this test will pass even if you remove the `error.type === 'can-type-error'` check.